common/app/common/akka.scala

```diff
@@ -3,31 +3,38 @@ package common
 import akka.agent.Agent
 import play.api.libs.concurrent.{Akka => PlayAkka}
 import scala.concurrent.duration._
-import play.api.Play
+import play.api.{Mode, Environment, Play}
 import scala.concurrent.ExecutionContext
 import akka.actor.ActorSystem
-import play.api.Play.current
 
 object ExecutionContexts extends ExecutionContexts
 
 trait ExecutionContexts {
   implicit lazy val executionContext: ExecutionContext = play.api.libs.concurrent.Execution.Implicits.defaultContext
-  lazy val actorSystem: ActorSystem = PlayAkka.system
-  lazy val feedsRecorderExecutionContext: ExecutionContext = PlayAkka.system.dispatchers.lookup("akka.actor.feed-recorder")
+  lazy val actorSystem: ActorSystem = PlayAkka.system(Play.current)
+  lazy val feedsRecorderExecutionContext: ExecutionContext = PlayAkka.system(Play.current).dispatchers.lookup("akka.actor.feed-recorder")
 }
 
 object AkkaAgent extends ExecutionContexts {
   def apply[T](value: T): Agent[T] = Agent(value)
 }
 
-object AkkaAsync extends ExecutionContexts {
+class AkkaAsync(env: Environment, actorSystem: ActorSystem) {
+  implicit val ec: ExecutionContext = actorSystem.dispatcher
 
   def apply(body: => Unit): Unit = after(1.second){ body }
```

Comment:
Yes I found that using the apply method on a class doesn't bring much readability.
`AkkaAsync { blah }` => `akkaAsync { blah }`
For some reason it just looks dodgy, especially as you can name your object as you please.

Also, "akkaAsync.apply" isn't really self-explanatory about what it does, so yes I think the apply function should be removed. I would like to mark it as deprecated but that would raise a compilation error given our compilation options.